Framework/DataObjects/src/EventList.cpp

```diff
@@ -3937,6 +3953,31 @@ void EventList::filterInPlace(Kernel::SplittingIntervalVec &splitter) {
   }
 }
 
+/**
+ * Initialize the detector ID's and event type of the destination event lists when splitting this list.
+ * @param partials : resulting partial lists of events after splitting's done
+ */
+void EventList::initializePartials(std::map<int, EventList *> partials) const {
+
+  // collect the state from events which is to be transferred to the partials
+  bool removeDetIDs{true};
+  const auto &detIDs = this->getDetectorIDs();
+  auto histogram = this->getHistogram();
```

Comment:
should `histogram` and `eventType` be declared `const` so the compiler can optimize it away?